File: API/Helpers/BookParams.cs
namespace API.Helpers
{
    public class BookParams : PaginationParams
    {
        public string? Title { get; set; }
        public string? Author { get; set; }
        public string? Publisher { get; set; }
        public string? PublishingDate { get; set; }
        public string? Genre { get; set; }
        public string? Isbn { get; set; }
        public int? Status { get; set; }
    }
}

File: client/src/app/_models/bookParams.ts
import { Paged } from './paged';

export class BookParams extends Paged {
  title: string | null = '';
  author: string | null = '';
  publisher: string | null = '';
  publishingDate: Date | null = null;
  genre: string | null = '';
  isbn: string | null = '';
  status: number | null = 0;
}
